client/containers/Scenarios.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component, Fragment } from 'react'
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Input, ListGroup, ListGroupItem } from 'reactstrap'
+import ScenarioPrompt from '../components/ScenarioPrompt'
+
+const styles = {
+  scenariosList: {
+    height: '200px',
+    overflow: 'scroll'
+  },
+  scenario: {
+    cursor: 'pointer'
+  }
+}
+
+export default class Scenarios extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { 
+      modal: false,
+      nestedModal: false,
+      newScenarioName: null,
+      selectedScenario: null,
+      savedScenarios: []
+    }
+    this.toggle = this.toggle.bind(this)
+    this.toggleNested = this.toggleNested.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSave = this.handleSave.bind(this)
+    this.getScenarios = this.getScenarios.bind(this)
+    this.handleOpen = this.handleOpen.bind(this)
+    this.handleSelect = this.handleSelect.bind(this)
+  }
+
+  toggle() {
+    if (!this.state.modal) {
+      this.getScenarios()
+    }
+    else {
+      this.setState({ 
+        modal: !this.state.modal,
+        newScenarioName: null
+      })
+    }
+  }
+
+  toggleNested() {
+    this.setState({ nestedModal: !this.state.nestedModal })
+  }
+
+  handleChange({ target }) {
+    this.setState({ newScenarioName: target.value })
+  }
+
+  handleSave() {
+    const { newScenarioName } = this.state
+    if (!newScenarioName) return this.toggleNested()
+    const { inputs } = this.props
+    const reqBody = Object.assign({}, { name: newScenarioName }, inputs)
+    const req = {
+      method: 'POST',
+      body: JSON.stringify(reqBody),
+      headers: { 'Content-Type': 'application/json' }
+    }
+    fetch('/scenarios', req)
+      .then(res => res.ok ? res.json() : null)
+      .then(scenario => scenario && this.toggle())
+  }
+
+  handleSelect({ target }) {
+    const name = target.getAttribute('data-name')
+    const id = target.getAttribute('data-id')
```

Comment:
`data-` attributes are available on the element's `dataset`.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset
```js
const { id, name } = target.dataset
```